src/rust/ide/src/view/node_searcher.rs

```diff
@@ -80,6 +86,28 @@ impl NodeSearcher {
         self.display_object.add_child(&self.text_field.display_object());
         self.text_field.clear_content();
         self.text_field.set_focus();
+        let module     = self.node_editor.displayed_module();
+        //TODO[ao]: Now we use some predefined location, until this task will be done:
+        // https://github.com/enso-org/ide/issues/653 . This code should be replaced with
+        // the proper Searcher view integration anyway.
+        let position   = TextLocation { line:2, column:4 };
+        let controller = controller::Searcher::new(&self.logger,&*self.project,module,position);
+        let logger     = self.logger.clone_ref();
+        let weak       = Rc::downgrade(&self.controller);
+        executor::global::spawn(controller.subscribe().for_each(move |notification| {
+            if let Some(opt_controller) = weak.upgrade() {
+                if let Some(controller) = opt_controller.get() {
+                    match notification {
+                        controller::searcher::Notification::NewList => {
+                            let list = controller.suggestions();
+                            info!(logger,"New list in Searcher: {list:?}");
+                        }
+                    }
+                }
+            }
+            futures::future::ready(())
+        }));
```

Comment:
So we add new subscription each time when `show` is called.
Do we unsubscribe when hiding the searcher?

Reply:
Thanks for pointing out, that I forget to remove controller on hide. When removed, the notifications publisher should perish and the executed task should finish.